Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemos/GazeContingent/FixWindowBufferedSamples/EyeLink_FixWindowBufferedSamples.m

```diff
@@ -0,0 +1,538 @@
+    el.backgroundcolour = [128 128 128];% RGB grey
+    el.calibrationtargetcolour = [0 0 0];% RGB black
+    % set "Camera Setup" instructions text colour so it is different from background colour
+    el.msgfontcolour = [0 0 0];% RGB black
+    % You must call this function to apply the changes made to the el structure above
+    EyelinkUpdateDefaults(el);
+    
+    % Set display coordinates for EyeLink data by entering left, top, right and bottom coordinates in screen pixels
+    Eyelink('Command','screen_pixel_coords = %ld %ld %ld %ld', 0, 0, width-1, height-1);
+    % Write DISPLAY_COORDS message to EDF file: sets display coordinates in DataViewer
+    % See DataViewer manual section: Protocol for EyeLink Data to Viewer Integration > Pre-trial Message Commands
+    Eyelink('Message', 'DISPLAY_COORDS %ld %ld %ld %ld', 0, 0, width-1, height-1);    
+    % Set number of calibration/validation dots and spread: horizontal-only(H) or horizontal-vertical(HV) as H3, HV3, HV5, HV9 or HV13
+    Eyelink('Command', 'calibration_type = HV9'); % horizontal-vertical 9-points
+    % Allow a supported EyeLink Host PC button box to accept calibration or drift-check/correction targets via button 5
+    Eyelink('Command', 'button_function 5 "accept_target_fixation"');
+    % Hide mouse cursor
+    HideCursor(screenNumber);
+    % Start listening for keyboard input. Suppress keypresses to Matlab windows.
+    ListenChar(-1);
+    Eyelink('Command', 'clear_screen 0'); % Clear Host PC display from any previus drawing
+    % Put EyeLink Host PC in Camera Setup mode for participant setup/calibration
+    EyelinkDoTrackerSetup(el);
+  
+    
+    %% STEP 5: TRIAL LOOP.
+    
+    % Create central square fixation window
+    fixationWindow = [-fixWinSize -fixWinSize fixWinSize fixWinSize];
+    fixationWindow = CenterRect(fixationWindow, wRect);
+    
+    spaceBar = KbName('space');% Identify keyboard key code for space bar to end each trial later on
+    imgList = {'img1.jpg' 'img2.jpg'};% Provide image list for 2 trials
+    
+    for i = 1:length(imgList) % Trial loop        
+        % Reset some parameters for each trial
+        sCross = 0; % Reset crosshairs display marker for each trial
+        fixWinComplete = 'yes'; % Reset variable for gaze maintained inside fixation window successfully
+        
+        % STEP 5.1: PREBUILD STIMULUS (GREY BACKGROUND + IMAGE + TEXT)
+        
+        % Prepare grey background on backbuffer
+        Screen('FillRect', window, el.backgroundcolour);
+        % Use 'drawBuffer' to copy unprocessed backbuffer images without additional processing. Prevents image size info issues on Retina displays
+        backgroundArray = Screen('GetImage', window, [], 'drawBuffer'); % Copy unprocessed backbuffer
+        backgroundTexture = Screen('MakeTexture', window, backgroundArray); % Convert background to texture so it is ready for drawing later on       

```

Comment:
backgroundTexture seems to be only used to take a screenshot of a uniform/empty window in background color el.backgroundcolor? Why? Why not use `` Screen('FillRect', window, el.backgroundcolour);`` instead of Drawtexture to clear to background color in a much more efficient way?